packages/cli/internal/pkg/version/checker.go

```diff
@@ -0,0 +1,53 @@
+package version
+
+import (
+	"time"
+
+	"github.com/blang/semver/v4"
+)
+
+type checker struct {
+	store Store
+
+	currentTime time.Time
+}
+
+func (ch *checker) Check(versionFullString string) (Result, error) {
```

Comment:
The logic to compute the result here looks like it can be simplified. Thoughts on something like this?

```
func (ch *checker) Check(versionFullString string) (Result, error) {
  currentSemVersion, err := semver.Parse(versionFullString)
  if err != nil {
    return Result{}, err
  }
  finalizeVersion(&currentSemVersion)
  currentVersion := currentSemVersion.String()
  result := Result{CurrentVersion: currentVersion, LatestVersion: currentVersion}
  infos, err := ch.store.ReadVersions(currentVersion, ch.currentTime)
  if err != nil {
    return Result{}, err
  }

  startIndex := findStartIndex(infos, startVersion)

  for i := startIndex; i < len(infos); i++ {
    info := infos[i]
    if i == 0 && currentSemVersion.EQ(info.Version) {
      result.CurrentVersionDeprecated = info.Deprecated
      result.CurrentVersionDeprecationMessage = info.DeprecationMessage
      continue
    }
    if len(info.Highlight) > 0 {
      result.NewerVersionHighlights = append(result.NewerVersionHighlights, info.Highlight)
    }
    result.LatestVersion = info.Name
  }

  return result, nil 
}

func findStartIndex(infos []Info, startVersion semver.Version) int {
  return sort.Search(len(infos), func(i int) bool {
    return infos[i].Version.GTE(startVersion)
  })
}
```

Reply:
I'm not sure I understand what you mean by simplification in this case. You example seems more involved and has redundant binary search. We know that current version (if exists) is the first item in the list.